include/mbedtls/net_sockets.h

```diff
@@ -106,6 +106,23 @@ mbedtls_net_context;
  */
 void mbedtls_net_init( mbedtls_net_context *ctx );
 
+/**
+ * \brief          Set file descriptor for given context
```

Comment:
How is this function supposed to behave on Windows?

On Windows, socket handles are not file descriptors. The `fd` field of `mbedtls_net_context` contains a socket handle, not a file descriptor. Functions that use the context expect a socket handle.

The behavior of this function on Windows needs to be documented. It could be: function not available at build time, or function returning an error at runtime, or function expecting a socket handle (I don't like this: it's confusing since the function's name strongly implies that it expects a file descriptor), or function converting the given file descriptor to a socket handle (is this even possible?).

Reply:
In OBS Studio case ([here](https://github.com/obsproject/obs-studio/blob/master/plugins/obs-outputs/librtmp/rtmp.c#L866)) they use the WSASocket which seem to [return](https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-wsasocketa#return-value) a descriptor and then (on any platform) they put it in `mbedtls_net_context` `fd`.

So the behavior should be the same on both Linux and Windows.

Since `fd` became private since version 3, I'm just trying to add at least a setter for it.

If it re-became public thanks to #4842, then this PR is no longer needed.